pallets/moonbeam-lazy-migrations/src/tests.rs

```diff
@@ -16,10 +16,254 @@
+				RuntimeOrigin::signed(AccountId32::from([45; 32])),
+				vec![contract1_address].try_into().unwrap(),
+				1000
+			),
+			Error::<Runtime>::ContractNotCorrupted
+		);
+
+		assert_noop!(
+			LazyMigrations::clear_suicided_storage(
+				RuntimeOrigin::signed(AccountId32::from([45; 32])),
+				vec![contract2_address].try_into().unwrap(),
+				1000
+			),
+			Error::<Runtime>::ContractNotCorrupted
+		);
+
+		// Check that no storage has been removed
+
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract1_address).count(),
+			10
+		);
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract2_address).count(),
+			10
+		);
+	})
+}
+
+// Test that the extrinsic can handle an empty input
+#[test]
+fn test_clear_suicided_empty_input() {
+	ExtBuilder::default().build().execute_with(|| {
+		let contract_address = mock_contract_with_entries(1, 1, 10);
+
+		let _ = LazyMigrations::clear_suicided_storage(
+			RuntimeOrigin::signed(AccountId32::from([45; 32])),
+			vec![].try_into().unwrap(),
+			1000,
+		);
+
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract_address).count(),
+			10
+		);
+	})
+}
+
+// Test with multiple deleted contracts ensuring that the extrinsic can handle
+// multiple addresses at once.
+#[test]
+fn test_clear_suicided_contract_multiple_addresses() {
+	ExtBuilder::default().build().execute_with(|| {
+		let contract_address1 = mock_contract_with_entries(1, 1, 10);
+		let contract_address2 = mock_contract_with_entries(2, 1, 20);
+		let contract_address3 = mock_contract_with_entries(3, 1, 30);
+
+		// Call the extrinsic to delete the storage entries
+		let _ = LazyMigrations::clear_suicided_storage(
+			RuntimeOrigin::signed(AccountId32::from([45; 32])),
+			vec![contract_address1, contract_address2, contract_address3]
+				.try_into()
+				.unwrap(),
+			1000,
+		)
+		.unwrap();
+
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract_address1).count(),
+			0
+		);
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract_address2).count(),
+			0
+		);
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract_address3).count(),
+			0
+		);
+	})
+}
+
+// Test that the limit of entries to be deleted is respected
+#[test]
+fn test_clear_suicided_entry_limit() {
+	ExtBuilder::default().build().execute_with(|| {
+		let contract_address1 = mock_contract_with_entries(1, 1, 2000);
+		let contract_address2 = mock_contract_with_entries(2, 1, 1);
+
+		let _ = LazyMigrations::clear_suicided_storage(
+			RuntimeOrigin::signed(AccountId32::from([45; 32])),
+			vec![contract_address1, contract_address2]
+				.try_into()
+				.unwrap(),
+			1000,
+		)
+		.unwrap();
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract_address1).count(),
+			1000
+		);
+
+		assert_eq!(
+			pallet_evm::AccountStorages::<Runtime>::iter_prefix(contract_address2).count(),
+			1
+		);
+	})
```

Comment:
Maybe for the sake of completeness we could repeat the call. So that it stops right after removing `contract_address1` then check suicided contracts was incremented.

Reply:
That is already tested on line 89. test: `test_clear_suicided_contract_succesfull`